src/mintpy/utils/utils0.py

```diff
@@ -259,21 +259,25 @@ def touch(fname_list, times=None):
 ################################## Coordinate ##########################################
 def utm_zone2epsg_code(utm_zone):
     """Convert UTM Zone string to EPSG code.
-    Parameters: utm_zone - str, atr['UTM_ZONE']
-    Returns:    epsg     - str, EPSG code
-    Examples:   epsg = utm_zone2epsg_code('11N')
+
+    Parameters: utm_zone  - str, atr['UTM_ZONE']
+    Returns:    epsg_code - str, EPSG code
+    Examples:   epsg_code = utm_zone2epsg_code('11N')
     """
     from pyproj import CRS
-    crs = CRS.from_dict({'proj': 'utm',
-                         'zone': int(utm_zone[:-1]),
-                         'south': utm_zone[-1] == 'S',
-                        })
-    epsg = crs.to_authority()[1]
-    return epsg
+
+    # N is the first letter in the northern hemisphere
+    crs = CRS.from_dict({
+        'proj': 'utm',
+        'zone': int(utm_zone[:-1]),
+        'south': utm_zone[-1].upper() < 'N',
```

Comment:
Is `utm_zone[-1].upper()` either `"S"` or `"N"`? If so, then this is always false:

```
>>> 'N' < 'N'
False
>>> 'S' < 'N'
False
```

Reply:
Oh sorry, this is for the "UTM Zone Designators" from here? http://www.jaworski.ca/utmzones.htm I had only seen them called "Latitude Bands" as part of MGRS https://en.wikipedia.org/wiki/Universal_Transverse_Mercator_coordinate_system#Latitude_bands

Reply:
I should have asked for your review here. The `UTM_ZONE` attribute is grabbed using `gdal` [here](https://github.com/insarlab/MintPy/blob/308024a83f7baa06b8532c11a1496195fca28280/src/mintpy/utils/readfile.py#L1653). So I am not sure if the last letter is the zone designator (latitude band), or just N / S. The change here assumes the zone designator.

TBH, I don't like the above two, which can be confusing. After searching gdal, the [`GetUTMZone()`](https://gdal.org/doxygen/classOGRSpatialReference.html#abdbd0df4b5ade0dded86d8a4729e4019) seems like a nice solution, with a negative value for the southern hemisphere. Shall we use this convention instead for `UTM_ZONE` attribute? How does OPERA handle this?

Reply:
I think we've been trying to just use EPSG codes, since they are more general than UTM and are just as easy to use with `pyproj` or gdal to warp coordinates due to the nice `from_user_input` and `from_epsg`:

```python
In [1]: from pyproj import CRS
# e.g. the EPSG code 32617 is zone 17N
In [2]: CRS.from_epsg(32617) == CRS.from_user_input("epsg:32617") == CRS.from_user_input("32617")
Out[2]: True

In [3]: CRS.from_epsg(32617)
Out[3]:
<Derived Projected CRS: EPSG:32617>
Name: WGS 84 / UTM zone 17N
Axis Info [cartesian]:
- E[east]: Easting (metre)
- N[north]: Northing (metre)
Area of Use:
- name: Between 84°W and 78°W, northern hemisphere between equator and 84°N, onshore and offshore. Bahamas. Ecuador - north of equator. Canada - Nunavut; Ontario; Quebec. Cayman Islands. Colombia. Costa Rica. Cuba. Jamaica. Nicaragua. Panama. United States (USA).
- bounds: (-84.0, 0.0, -78.0, 84.0)
Coordinate Operation:
- name: UTM zone 17N
- method: Transverse Mercator
Datum: World Geodetic System 1984 ensemble
- Ellipsoid: WGS 84
- Prime Meridian: Greenwich
```

I'm now seeing how the confusion arose... it's very unfortunate that the `utm` python package links to  http://www.jaworski.ca/utmzones.htm , since I'm >90% sure those "Zone designators" are not part of UTM, but rather MGRS:

https://cove.army.gov.au/article/smart-soldier-understanding-military-grid-reference-system

>A UTM coordinate comprises a zone number, a hemisphere (N/S), an easting and a northing. [....]

>UTM zones are divided in the same way MGRS zones are divided, although there is no letter designation after the zone number. Instead this letter indicator is replaced with an ‘N’ or ‘S’ depending on which hemisphere the map is within.

MGRS is built on top of UTM, so the initial zones are the same... but there's not the latitude bands, just "north/south". I think MGRS the added extra specifiers as a way to quickly communicate locations on maps.

Reply:
I see, thanks for the examples and links. I will use `EPSG` code then as the primary metadata; and the `UTM_ZONE` without the latitude band as the secondary metadata, while converting coordinates. 

I will fix the `utm_zone2epsg_code()` with a PR right away.

It seems that `pyproj` may be enough for the coordinate conversion, and we do not really need the `utm` package (although it seems faster than pyproj but the speed is not our concern here since the conversion will be rare). Converting UTM to lat/lon seems straightforward, as a similar function already exists ([utils.utils0.to_latlon()](https://github.com/insarlab/MintPy/blob/308024a83f7baa06b8532c11a1496195fca28280/src/mintpy/utils/utils0.py#L279)). 

I will work on the function to convert lat/lon to UTM when I find time, the unknown is how to calculate the epsg code given a lat/lon. If you have the solution, please feel free to go for it as well.